src/foam/parse/EnumParserFactory.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.parse;
+
+import java.util.HashMap;
+
+import foam.lib.json.IntParser;
+import foam.lib.json.Whitespace;
+import foam.lib.parse.Alt;
+import foam.lib.parse.Literal;
+import foam.lib.parse.PStream;
+import foam.lib.parse.Parser;
+import foam.lib.parse.ParserContext;
+import foam.lib.parse.Repeat0;
+import foam.lib.parse.Seq0;
+
+public class EnumParserFactory {
+  private static HashMap<Class, Parser> parsers_ = new HashMap<Class, Parser>();
+
+  public static Parser getInstance(Class c) {
+    if ( parsers_.containsKey(c) ) {
+      return parsers_.get(c);
+    }
+
+    Parser parser = buildInstance_(c);
+    parsers_.put(c, parser);
+    return parser;
+  }
+
+  public static Parser buildInstance_(Class c) {
+    return new Repeat0(new Seq0(

```

Comment:
I think Enum values for queries will need a very different parser.  If we have a situation for

foam.ENUM({
  name: 'FooEnum',
  values: [
    'FOO',
    'BAR'
  ]
});

foam.CLASS({
  name: 'Foo',
  properties: [
    { class: 'Enum', of: 'FooEnum' name: 'foo' }
  ]
});
```

We'll want to parse queries like

foo:BAR or -foo:FOO

This looks like it's trying to parser Enum values as JSON.